cmds/db-manager/main.go

```diff
@@ -0,0 +1,219 @@
+type Direction int
+
+func (d Direction) String() string {
+	if d > 0 {
+		return "Up"
+	} else if d < 0 {
+		return "Down"
+	}
+	return "No Change"
+}
+
+var (
+	path      = flag.String("migration_dir", "", "path to db migration files directory")
+	dbVersion = flag.String("db_version", "", "the db version to migrate to (ex: v1.0.0)")
+	step      = flag.Int("migration_step", 0, "the db migration step to go to")
+
+	cockroachParams = struct {
+		host            *string
+		port            *int
+		sslMode         *string
+		sslDir          *string
+		user            *string
+		applicationName *string
+	}{
+		host:    flag.String("cockroach_host", "0.0.0.0", "cockroach host to connect to"),
+		port:    flag.Int("cockroach_port", 26257, "cockroach port to connect to"),
+		sslMode: flag.String("cockroach_ssl_mode", "disable", "cockroach sslmode"),
+		user:    flag.String("cockroach_user", "root", "cockroach user to authenticate as"),
+		sslDir:  flag.String("cockroach_ssl_dir", "", "directory to ssl certificates. Must contain files: ca.crt, client.<user>.crt, client.<user>.key"),
+	}
+)
+
+func main() {
+	flag.Parse()
+	if *path == "" {
+		log.Panic("Must specify migration_dir path")
+	}
+	if (*dbVersion == "" && *step == 0) || (*dbVersion != "" && *step != 0) {
+		log.Panic("Must specify a db_version or migration_step to goto")
+	}
+	uriParams := map[string]string{
+		"host":             *cockroachParams.host,
+		"port":             strconv.Itoa(*cockroachParams.port),
+		"user":             *cockroachParams.user,
+		"ssl_mode":         *cockroachParams.sslMode,
+		"ssl_dir":          *cockroachParams.sslDir,
+		"application_name": "SchemaManager",
+	}
+	postgresURI, err := dssCockroach.BuildURI(uriParams)
+	if err != nil {
+		log.Panic("Failed to build URI", zap.Error(err))
+	}
+	crdbURI := strings.Replace(postgresURI, "postgresql", "cockroachdb", 1)
+	*path = fmt.Sprintf("file://%v", *path)
+	migrater, migraterErr := migrate.New(*path, crdbURI)
+	myMigrater := &MyMigrate{migrater}
+	defer func() {
+		if migraterErr == nil {
+			if _, err := myMigrater.Close(); err != nil {
+				log.Println(err)
+			}
+		}
+	}()
+	if migraterErr == nil {
+		myMigrater.PrefetchMigrations = 10
+		myMigrater.LockTimeout = time.Duration(15) * time.Second
+
+		// handle Ctrl+c
+		signals := make(chan os.Signal, 1)
+		signal.Notify(signals, syscall.SIGINT)
+		go func() {
+			for range signals {
+				log.Println("Stopping after this running migration ...")
+				myMigrater.GracefulStop <- true
+				return
+			}
+		}()
+	}
+	totalMove := 0
+	migrateDirection := myMigrater.MigrationDirection(postgresURI, *dbVersion, *step)
+	for migrateDirection != 0 {
+		migraterErr = myMigrater.Steps(int(migrateDirection))
+		if migraterErr != nil {
+			log.Fatal(migraterErr)
+		}
+		totalMove += int(migrateDirection)
+		log.Printf("Migrated %s by %d step", migrateDirection.String(), int(math.Abs(float64(int(migrateDirection)))))
```

Comment:
Let's record myMigrater.Version() before calling .Steps and after calling .Steps and report that version transition in this log statement.